python/pdo/submitter/ccf/ccf_submitter.py

```diff
@@ -0,0 +1,589 @@
+            nonce = channel_keys
+            )
+
+        try:
+            response = self.submit_rpc_to_ccf(tx_method, tx_params)
+            if response.result: # result will be True for successful init transaction
+                return tx_params['nonce'] #this will represent the transaction id
+            else:
+                raise Exception(response.error['message'])
+        except Exception as e:
+            raise
+
+# -----------------------------------------------------------------
+    def get_enclave_info(self,
+        enclave_id):
+
+        tx_method = "verify_enclave_registration"
+        tx_params = PayloadBuilder.build_verify_enclave_from_data(enclave_id)
+
+        enclave_info = self.submit_read_request_to_ccf(tx_method, tx_params)
+
+        # verify ccf signature
+        message = enclave_info["verifying_key"]
+        message+= enclave_info["encryption_key"]
+        message+= enclave_info["proof_data"]
+        message+= enclave_info["last_registration_block_context"]
+        message+= enclave_info["owner_id"]
+
+        if not CCFSubmitter.verify_ledger_signature(message, enclave_info["signature"]):
+            raise Exception("Invalid signature on Get Enclave Info from CCF Ledger")
+
+        return enclave_info
+
+# -----------------------------------------------------------------
+    def get_contract_info(self,
+        contract_id):
+
+        tx_method = "verify_contract_registration"
+        tx_params = PayloadBuilder.build_verify_contract_registration_from_data(contract_id)
+
+        contract_info = self.submit_read_request_to_ccf(tx_method, tx_params)
+
+        # verify ccf signature
+        message = contract_info["contract_id"]
+        message+= contract_info["contract_code_hash"]
+        message+= contract_info["pdo_contract_creator_pem_key"]
+        for pservice_id in contract_info["provisioning_service_ids"]:
+            message+= pservice_id
+        message+= contract_info["enclaves_info"]
+
+        if not CCFSubmitter.verify_ledger_signature(message, contract_info["signature"]):
+            raise Exception("Invalid signature on Get Contract Info from CCF Ledger")
+
+        return contract_info
+
+# -----------------------------------------------------------------
+    def get_current_state_hash(self,
+        contract_id):
+
+        tx_method = "get_current_state_info_for_contract"
+        tx_params = PayloadBuilder.build_get_current_state_info_for_contract_from_data(contract_id)
+
+        state_info = self.submit_read_request_to_ccf(tx_method, tx_params)
+
+         # verify ccf signature
+        message = contract_id + state_info["state_hash"]
+
+        if not CCFSubmitter.verify_ledger_signature(message, state_info["signature"]):
+            raise Exception("Invalid signature on Get Current State Hash from CCF Ledger")
+
+        return state_info
+
+# -----------------------------------------------------------------
+    def get_state_details(self,
```

Comment:
In what context would this function be called? I'm wondering if we can give it a more meaningful name that would reflect that.

Reply:
thank you for the question. Used before submitting a transaction to check if dependencies are met. (see transaction.py, Dependency class). I am happy to rename if there is something more apt.

Reply:
Actually, to keep the APIs consistent, we would have to change the name in sawtooth as well, so let's keep it for now.